astropy/io/fits/tests/test_core.py

```diff
@@ -299,13 +295,8 @@ def make_invalid_hdu():
 
         # fix+exception
         hdu = make_invalid_hdu()
-        try:
+        with pytest.raises(fits.VerifyError , match=r'(?s)not upper case..+ Illegal keyword name') as excinfo:
```

Comment:
Not sure what's up with the `?s`, the `..` and the `+`. I would prefer the old-style `in` checks. Also no reason to use the `as excinfo` if you decide to keep this style.

Reply:
Admittedly I've abused this as a bit of a playground/test bed for how complex a match I could implement just with `regex`s. The `(?s)` as a flag for accepting linebreaks in the string is described in the `testguide.rst` entry. In the first test I had to keep the extras `assert ... in`, because the non-occurrence cannot be tested within a single regex (or at least not without making it even more convoluted ;-).
But for practical purposes the two separate checks are probably far more readable, so I can revert the second test as well, and just keep the example in the docs (unless we decide we want to discourage too complex expressions altogether).

Reply:
Ah thanks for the explanation. So the `..+` checks if there are at least two characters after the `case`? 😄 Wouldn't `.{2,}` be more appropriate then?

Reply:
> But for practical purposes the two separate checks are probably far more readable, so I can revert the second test as well

I would prefer that.

Reply:
:+1: to keep things simple, `assert ... in` is more readable here.